monai/data/box_utils.py

```diff
@@ -572,3 +572,261 @@ def convert_box_to_standard_mode(
             box_convert_mode(boxes=boxes, src_mode="xxyyzz", dst_mode="xyzxyz")
     """
     return convert_box_mode(boxes=boxes, src_mode=mode, dst_mode=StandardMode())
+
+
+def check_boxes(boxes: NdarrayOrTensor) -> bool:
+    """
+    This function checks whether the box size is non-negative.
+
+    Args:
+        boxes: bounding box, Nx4 or Nx6 torch tensor or ndarray. The box mode is assumed to be ``StandardMode``
+
+    Returns:
+        whether ``boxes`` is valid
+    """
+    spatial_dims = get_spatial_dims(boxes=boxes)
+    for axis in range(0, spatial_dims):
+        if (boxes[:, spatial_dims + axis] < boxes[:, axis]).sum() > 0:
+            return False
+    return True
+
+
+def box_area(boxes: NdarrayOrTensor) -> NdarrayOrTensor:
+    """
+    This function computes the area of each box
+
+    Args:
+        boxes: bounding box, Nx4 or Nx6 torch tensor or ndarray. The box mode is assumed to be ``StandardMode``
+
+    Returns:
+        area of boxes, with size of (N,).
+    """
+
+    if not check_boxes(boxes):
+        raise ValueError("Given boxes has invalid values. The box size must be non-negative.")
+
+    spatial_dims = get_spatial_dims(boxes=boxes)
+
+    area = boxes[:, spatial_dims] - boxes[:, 0] + TO_REMOVE
+    for axis in range(1, spatial_dims):
+        area = area * (boxes[:, axis + spatial_dims] - boxes[:, axis] + TO_REMOVE)
+
+    # convert numpy to tensor if needed
+    area_t, *_ = convert_data_type(area, torch.Tensor)
+
+    # check if NaN or Inf, especially for half precision
+    if area_t.isnan().any() or area_t.isinf().any():
+        if area_t.dtype is torch.float16:
+            raise ValueError("Box area is NaN or Inf. boxes is float16. Please change to float32 and test it again.")
+        else:
+            raise ValueError("Box area is NaN or Inf.")
+
+    # convert tensor back to numpy if needed
+    area, *_ = convert_to_dst_type(src=area_t, dst=area)
```

Comment:
I think you converted the `area` to `area_t` just for a check, so here we can return the previous `area` directly?

Thanks.